### PR TITLE
refactor(ayc): api-client revamp

### PR DESCRIPTION
**Pull request by lukeromanowicz**

### CODE REVIEW

File: packages/about-you/api-client/package.json [hunk 1]
```diff
@@ -1,19 +1,19 @@
 {
   "name": "@vue-storefront/about-you-api",
-  "version": "0.0.1",
+  "version": "0.1.0",
   "private": true,
   "sideEffects": false,
   "main": "lib/index.cjs.js",
   "module": "lib/index.es.js",
-  "types": "lib/index.d.ts",
+  "types": "lib/about-you/api-client/src/index.d.ts",
```
Comment (krskibin) [inline packages/about-you/api-client/package.json:8]: Is this path correct? Rest of the index files are in lib/index*.

File: packages/about-you/composables/src/composables/useProduct/index.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+import { getProductsByQuery } from '@vue-storefront/about-you-api';
 
 const useProduct: (cacheId: string) => UseProduct<BapiProduct> = useProductFactory<BapiProduct, any>({
-  productsSearch: mapProductSearch
+  productsSearch: async (params) => {
+    const products = await getProductsByQuery(mapProductSearchByQueryParams(params));
+
+    return {
+      data: products.entities,
+      total: products.pagination?.total || products.entities.length
```
Comment (krskibin) [inline packages/about-you/composables/src/composables/useProduct/index.ts:13]:
````
total: products.pagination?.total || products.entities?.length
````

File: packages/about-you/composables/src/helpers/product/mapProductSearchByQueryParams.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { ProductSearchQuery } from '@aboutyou/backbone/types/ProductSearchQuery';
+import { ProductsSearchEndpointParameters } from '@aboutyou/backbone/endpoints/products/products';
+
+const mapProductSearchByQueryParams = (params): ProductsSearchEndpointParameters => {
+  const searchQuery: ProductSearchQuery = {};
```
Comment (krskibin) [inline packages/about-you/composables/src/helpers/product/mapProductSearchByQueryParams.ts:5]:
```

const searchQuery = { categoryId: params.catId, term: params.term}
```

File: packages/about-you/api-client/package.json (hunk 1 above)
Comment (lukeromanowicz) [reply packages/about-you/api-client/package.json:8]: Yes, for some reason it is. I double-checked it with @filrak

Comment (krskibin) [reply packages/about-you/api-client/package.json:8]: Well. Ok, then

**Pull request merged**